### PR TITLE
Rac 186 genrate

### PR DESCRIPTION
**Pull request by juliensnz**
**Description (for Contributor and Core Developer)**

This PR adds the ability to generate a pull request with updated visuals if the visual regression tests are failing.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: .github/workflows/dsm-test.yml [hunk 1]
```diff
@@ -19,8 +19,38 @@ jobs:
...
         run: cd akeneo-design-system && yarn lint:check
-      - name: Launch tests
+      - name: Launch unit tests
         run: cd akeneo-design-system && yarn test:run
+      - name: Launch visual tests
+        run: cd akeneo-design-system && yarn test:visual:run
+        id: visual_test_execution
+      - name: Update visual tests
+        if: failure() && steps.visual_test_execution.outcome == 'failure'
+        run: cd akeneo-design-system && find . -name '*__image_snapshots__*'  -exec rm -rf {} \; || true && yarn test:visual:run -u
```
Comment (SamirBoulil) [inline .github/workflows/dsm-test.yml:29]: What's the "|| true" part for ? Maybe we should write a short comment to explain ?

File: .github/workflows/dsm-test.yml [hunk 2]
```diff
@@ -19,8 +19,38 @@ jobs:
...
+        uses: peter-evans/create-pull-request@v3
+        with:
+          commit-message: Update visual tests for branch ${{ steps.extract_branch.outputs.branch }}
+          branch: ${{ steps.extract_branch.outputs.branch }}-visual-update
+          base: ${{ steps.extract_branch.outputs.branch }}
+          title: Update visual tests for branch ${{ steps.extract_branch.outputs.branch }}
+          body: This PR updates the visual tests for the changes in the branch ${{ steps.extract_branch.outputs.branch }}
+          assignees: ${{ github.actor }}
+      - name: Comment PR
+        uses: thollander/actions-comment-pull-request@master
```
Comment (SamirBoulil) [inline .github/workflows/dsm-test.yml:47]: Should we use a tag for this action ?

File: .github/workflows/dsm-test.yml [hunk 3]
```diff
@@ -19,8 +19,38 @@ jobs:
...
-      - name: Launch tests
+      - name: Launch unit tests
         run: cd akeneo-design-system && yarn test:run
+      - name: Launch visual tests
+        run: cd akeneo-design-system && yarn test:visual:run
+        id: visual_test_execution
+      - name: Update visual tests
+        if: failure() && steps.visual_test_execution.outcome == 'failure'
+        run: cd akeneo-design-system && find . -name '*__image_snapshots__*'  -exec rm -rf {} \; || true && yarn test:visual:run -u
+      - uses: mdecoleman/pr-branch-name@1.0.0
```
Comment (SamirBoulil) [inline .github/workflows/dsm-test.yml:30]: is it a tag or branch ? should we use a tag ?

**Review by SamirBoulil — APPROVED** [review]

**Review by StevenVAIDIE — APPROVED** [review]

**Review by ValentinMumble — APPROVED** [review]

File: .github/workflows/dsm-test.yml (hunk 1 above)
Comment (juliensnz) [reply .github/workflows/dsm-test.yml:29]: Done!

**Pull request merged**